application-ckeditor-ui/src/main/resources/CKEditor/MacroService.xml

```diff
@@ -272,7 +272,10 @@
       #set ($translatedParameterDescriptor.defaultValue = $parameterDescriptor.defaultValue == 'true')
       #set ($translatedParameterDescriptor.type = $translatedParameterDescriptor.defaultValue.getClass())
     #end
-    #set ($htmlDisplayerParameters = {'name': $parameterDescriptor.id})
+    #set ($htmlDisplayerParameters = {
+      'name': $parameterDescriptor.id,
+      'sourceDocument': "$!request.sourceDocument"
```

Comment:
I see below that we set another parameter as ``data-property-group``, so I'm wondering if we shouldn't do the same for the source document. It wouldn't be nice to end up with a ``sourceDocument`` attribute in the HTML of the picker.  ``data-source-document`` would be better. I know you need this parameter only on the server-side, but other displayers might pass it to the client-side as an attribute.

The alternative is to change the way we call the macro service:

```
var sourceDocumentReference = XWiki.Model.resolve(editor.element.getAttribute('data-sourceDocumentReference'),
  XWiki.EntityType.DOCUMENT, XWiki.currentDocument.documentReference);
var url = new XWiki.Document(sourceDocumentReference).getURL('get', $.param({
  outputSyntax: 'plain',
  language: $('html').attr('lang'),
  sheet: 'CKEditor.MacroService'
}));
```

This is probably cleaner than passing the source document in lots of places.